pkg/virt-operator/webhooks/kubevirt-update-admitter.go

```diff
@@ -310,3 +323,32 @@ func validateInfraReplicas(replicas *uint8) []metav1.StatusCause {
 
 	return statuses
 }
+
+func toVerifyFeatureGates(currKVSpec, newKVSpec *v1.KubeVirtSpec) bool {
+	currDevConfig := currKVSpec.Configuration.DeveloperConfiguration
+	newDevConfig := newKVSpec.Configuration.DeveloperConfiguration
+
+	if (currDevConfig == nil && newDevConfig == nil) || (currDevConfig != nil && newDevConfig == nil) {
+		return false
+	}
+
+	if currDevConfig == nil && newDevConfig != nil {
+		return len(newDevConfig.FeatureGates) > 0
+	}
+
+	return !equality.Semantic.DeepEqual(currDevConfig.FeatureGates, newDevConfig.FeatureGates)
+}
+
+func warnDeprecatedFeatureGates(featureGates []string, config *virtconfig.ClusterConfig) (warnings []string) {
+	for _, featureGate := range featureGates {
+		if config.IsFeatureGateDeprecated(featureGate) {
+			const warningPattern = "feature gate %s is deprecated, therefore it can be safely removed and is redundant. " +
```

Comment:
Why don't you compose the warning message once and then use it twice instead of composing it twice and using it twice? `Waningf()` uses `Sprintf()` again under the hood.

Reply:
@acardace Sorry can you please give an example by what you mean?

Reply:
I also am not sure what you mean.
Bear in mind that `const`s are created once at compile-time.

Reply:
I mean doing like this
```go
for _, featureGate := range featureGates {
		if config.IsFeatureGateDeprecated(featureGate) {
			warningMsg := fmt.Sprintf("feature gate %s is deprecated, therefore it can be safely removed and is redundant. "+
				"For more info, please look at: https://github.com/kubevirt/kubevirt/blob/main/docs/deprecation.md", featureGate)
			warnings = append(warnings, warningMsg)

			log.Log.Warning(warningMsg)
		}
	}
```

Reply:
You can ignore me this is just a silly optimization but it saves you from doing 1 more memory allocation per featureGate as with your code you're calling into `fmt.Sprintf()` twice (which allocates memory).